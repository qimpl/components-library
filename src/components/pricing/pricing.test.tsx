import React from 'react';

import { render } from '@testing-library/react';

import Anchor from 'components/anchor';
import Button from 'components/button';
import Pricing from 'components/pricing';

test('Simple pricing block', () => {
  expect(
    render(
      <Pricing
        title='Business'
        subTitle='Sub title'
        features={['Feature 1', 'Feature 2', 'Feature 3', 'Feature 4', 'Feature 5', 'Feature 6']}
        price={15}
        buttonElement={
          <Anchor title='Subscribe' noUnderline>
            <Button>Subscribe</Button>
          </Anchor>
        }
      />
    )
  ).toMatchSnapshot();
});
