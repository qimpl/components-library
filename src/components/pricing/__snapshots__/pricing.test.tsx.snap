// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple pricing block 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fujyAs jUKVZh"
      >
        <h3
          class="sc-gKAaRy JtzLq"
          font-size="16px"
        >
          Business
        </h3>
        <h4
          class="sc-iCoGMd hkSkto"
          font-size="14px"
        >
          Sub title
        </h4>
        <ul>
          <li>
            <span>
              <svg
                aria-hidden="true"
                fill="#378a42"
                focusable="false"
                height="24"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
                />
              </svg>
            </span>
            <p
              class="sc-dlnjwi qSnYd"
              style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
            >
              Feature 1
            </p>
          </li>
          <li>
            <span>
              <svg
                aria-hidden="true"
                fill="#378a42"
                focusable="false"
                height="24"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
                />
              </svg>
            </span>
            <p
              class="sc-dlnjwi qSnYd"
              style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
            >
              Feature 2
            </p>
          </li>
          <li>
            <span>
              <svg
                aria-hidden="true"
                fill="#378a42"
                focusable="false"
                height="24"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
                />
              </svg>
            </span>
            <p
              class="sc-dlnjwi qSnYd"
              style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
            >
              Feature 3
            </p>
          </li>
          <li>
            <span>
              <svg
                aria-hidden="true"
                fill="#378a42"
                focusable="false"
                height="24"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
                />
              </svg>
            </span>
            <p
              class="sc-dlnjwi qSnYd"
              style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
            >
              Feature 4
            </p>
          </li>
          <li>
            <span>
              <svg
                aria-hidden="true"
                fill="#378a42"
                focusable="false"
                height="24"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
                />
              </svg>
            </span>
            <p
              class="sc-dlnjwi qSnYd"
              style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
            >
              Feature 5
            </p>
          </li>
          <li>
            <span>
              <svg
                aria-hidden="true"
                fill="#378a42"
                focusable="false"
                height="24"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
                />
              </svg>
            </span>
            <p
              class="sc-dlnjwi qSnYd"
              style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
            >
              Feature 6
            </p>
          </li>
        </ul>
        <div>
          <p
            class="sc-dlnjwi hqYSTK"
            style="color: rgb(1, 60, 99); font-size: 18px; line-height: 24px;"
          >
            15
            â‚¬ / Mois*
          </p>
          <a
            class="sc-bdnxRM gGfAxZ"
            href="#"
            target="_self"
            title="Subscribe"
          >
            <button
              class="sc-gtsrHT iPNkwx"
              color="#013c63"
              type="button"
            >
              Subscribe
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fujyAs jUKVZh"
    >
      <h3
        class="sc-gKAaRy JtzLq"
        font-size="16px"
      >
        Business
      </h3>
      <h4
        class="sc-iCoGMd hkSkto"
        font-size="14px"
      >
        Sub title
      </h4>
      <ul>
        <li>
          <span>
            <svg
              aria-hidden="true"
              fill="#378a42"
              focusable="false"
              height="24"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
              />
            </svg>
          </span>
          <p
            class="sc-dlnjwi qSnYd"
            style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
          >
            Feature 1
          </p>
        </li>
        <li>
          <span>
            <svg
              aria-hidden="true"
              fill="#378a42"
              focusable="false"
              height="24"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
              />
            </svg>
          </span>
          <p
            class="sc-dlnjwi qSnYd"
            style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
          >
            Feature 2
          </p>
        </li>
        <li>
          <span>
            <svg
              aria-hidden="true"
              fill="#378a42"
              focusable="false"
              height="24"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
              />
            </svg>
          </span>
          <p
            class="sc-dlnjwi qSnYd"
            style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
          >
            Feature 3
          </p>
        </li>
        <li>
          <span>
            <svg
              aria-hidden="true"
              fill="#378a42"
              focusable="false"
              height="24"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
              />
            </svg>
          </span>
          <p
            class="sc-dlnjwi qSnYd"
            style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
          >
            Feature 4
          </p>
        </li>
        <li>
          <span>
            <svg
              aria-hidden="true"
              fill="#378a42"
              focusable="false"
              height="24"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
              />
            </svg>
          </span>
          <p
            class="sc-dlnjwi qSnYd"
            style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
          >
            Feature 5
          </p>
        </li>
        <li>
          <span>
            <svg
              aria-hidden="true"
              fill="#378a42"
              focusable="false"
              height="24"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 15.9L4.7 10.6 3.6 11.6 8.9 16.9 10 18 20.6 7.4 19.5 6.3z"
              />
            </svg>
          </span>
          <p
            class="sc-dlnjwi qSnYd"
            style="color: rgb(67, 80, 88); font-size: 15px; line-height: 21px;"
          >
            Feature 6
          </p>
        </li>
      </ul>
      <div>
        <p
          class="sc-dlnjwi hqYSTK"
          style="color: rgb(1, 60, 99); font-size: 18px; line-height: 24px;"
        >
          15
          â‚¬ / Mois*
        </p>
        <a
          class="sc-bdnxRM gGfAxZ"
          href="#"
          target="_self"
          title="Subscribe"
        >
          <button
            class="sc-gtsrHT iPNkwx"
            color="#013c63"
            type="button"
          >
            Subscribe
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
